coralogix-policies/coralogix-metrics-integration-policy/template.yaml

```diff
@@ -76,14 +82,17 @@ Resources:
                 - "arn:aws:iam::${aws_account_id}:role/coralogix-ingestion-${role_suffix}"
                 - aws_account_id: !If
                     - IsCustomAccountId
-                    - !Ref CustomerAccountId
+                    - !Ref CustomAccountId
                     - !FindInMap [CoralogixEnvironment, !Ref AWSAccount, "ID"]
                   role_suffix: !FindInMap [CoralogixEnvironment, !Ref AWSAccount, "RoleSuffix"]
             Action:
               - sts:AssumeRole
             Condition:
               StringEquals:
-                sts:ExternalId: !Ref ExternalId
+                sts:ExternalId: !Sub
+                  - "${ExternalIdSecret}@${company_id}"
+                  - ExternalIdSecret: !Ref ExternalIdSecret
+                    company_id: !Ref CoralogixCompanyId
```

Comment:
I don't think we should do this, it defeats the purpose of having company_id in the external secret — we should use the full value of `ExternalIdSecret` in the condition.

Reply:
So where should we use the `company_id`? If we use the full `ExternalIdSecret` then we will still have the security issue